genie-agent/src/main/java/com/netflix/genie/agent/execution/services/impl/JobSetupServiceImpl.java

```diff
@@ -695,7 +695,7 @@ String composeScript() {
             sb
                 .append("# Launch the command")
                 .append(NEWLINE)
-                .append(this.commandLine).append(" &").append(NEWLINE)
+                .append(this.commandLine).append(" <&0 &").append(NEWLINE)
```

Comment:
What was reason in #1020 to turn
```bash
command
```
into
```bash
command &
wait %1
```

Reply:
https://github.com/Netflix/genie/pull/1020/commits/6a7c0d77c18d6bb950916e731f8f548860626e2c
Depending on the command, the bash process running `run` was terminating but leaving the underlying command running.
That change introduces a bash trap so that the job command is always terminated before a job is marked finished.